cmd/models/answer_option.model.go

```diff
@@ -0,0 +1,85 @@
+package models
+
+import (
+	"go-lms-of-pupilfirst/pkg/utils"
+)
+
+var (
+	answerOptionTableName = "answer_options"
+)
+
+// AnswerOption is a model for options of Answers for a target
+type AnswerOption struct {
+	utils.Base
+	QuizQuestionID string `sql:"type:uuid;" validate:"omitempty,uuid,required"`
+	Value          string
+	Hint           string `gorm:"type:varchar(255)"`
+
+	QuizQuestion *QuizQuestion `gorm:"foreignkey:QuizQuestionID"`
+}
+
+// TableName gorm standard table name
+func (a *AnswerOption) TableName() string {
+	return answerOptionTableName
+}
+
+// AnswerOptionList defines array of answer options objects
+type AnswerOptionList []*AnswerOption
```

Comment:
How come you're defining this list type everywhere?

Reply:
Eventually, I am going to need to fetch a list of records, so instead of creating the lists on the fly, I figured it would be convenient to define it once

Reply:
No need to do this IMO. Go already has a concise slice syntax.

Reply:
What do you mean? I am just defining a type beforehand since I am going to use it many times. I could always pass around []*Model on the fly... but with a defined type I can add more functions to it like filtering ...

Reply:
I just don't think you gain anything by having a type definition here. You can always write functions for `[]*AnswerOption` just like you can write functions for `AnswerOptionList`. 

The thing is that you're not doing anything with this type definition. Usually if you want to define a type alias (i.e., not a struct definition), you want to limit the scope of the accepted values for a function. For example, I could do something like this:

```
type IceCream string

const (
    Vanilla IceCream = "Vanilla"
    Chocolate IceCream = "Chocolate"
)

func GiveIceCream() IceCream {...}
```

to limit the strings returned by my function. But defining type aliases just like `type IceCream string` doesn't really do anything in my opinion.

Just my two cents, ultimately it's not a huge deal. Just doesn't really seem idomatic to me.

Reply:
I see. I will remove them.

Reply:
One other thing you can use type definitions to do is to formalize general data structures. For example, I could define `type ParameterMap map[string]string` if I'm using this specific concept in multiple places.

Reply:
yeah, actually that's why I defined the List types

Reply:
The difference is that `map[string]string` is general. If I'm defining `type ParameterMap map[string]string`, I'm saying "this is a map of my Parameters". 

IMO `type AnswerOptionList []*AnswerOption` doesn't add any extra information.

Again, this is stylistic, so it's up to you if you want to keep it!